src/searchModal.js

```diff
@@ -18,43 +18,51 @@
 import $ from 'jquery';
 import _ from 'lodash';
 import __ from 'i18n';
+import context from 'context';
 import layoutTpl from 'ui/searchModal/tpl/layout';
 import infoMessageTpl from 'ui/searchModal/tpl/info-message';
 import 'ui/searchModal/css/searchModal.css';
 import component from 'ui/component';
 import 'ui/modal';
 import 'ui/datatable';
+import store from 'core/store';
 
 /**
- * Creates a search modal instance
- * @param {object} config - search modal configuration
- * @param {string} config.query - search query to be set and triggered on component creation
- * @param {string} config.url - search endpoint
- * @param {object} config.events - events hub
- * @param {object} config.renderTo - dom element where component will be rendered
+ * Creates a searchModal instance
+ *
+ * @param {object} config
+ * @param {object} config.renderTo - DOM element where component will be rendered to
+ * @param {string} config.query - search query to be set on component creation
+ * @param {boolean} config.searchOnInit - if init search must be triggered or not (stored results are used instead)
+ * @param {string} config.url - search endpoint to be set on datatable
+ * @returns {searchModal}
  */
 export default function searchModalFactory(config) {
-    const defaults = {
+    config = _.defaults(config, {
```

Comment:
Please, check https://github.com/oat-sa/tao-core-ui-fe/blob/b4a4fdd3dd8eb797b092977061580250384cc548/src/component.js#L91-L97
you can pass `defaults` in `component()`
and passed configs can be sent to `init()`